pytorch_lightning/strategies/fully_sharded.py

```diff
@@ -135,16 +135,15 @@ def setup_distributed(self) -> None:
 
     def setup(self, trainer: "pl.Trainer") -> None:
         self.accelerator.setup(trainer)
-        self.setup_optimizers(trainer)
-        self.setup_precision_plugin()
-        optimizers_to_device(self.optimizers, self.root_device)
 
         if self._layer_sync:
             self.model = self._layer_sync.apply(self.model)
 
         self.configure_ddp()
         self.barrier()
         self.setup_optimizers(trainer)
+        optimizers_to_device(self.optimizers, self.root_device)
+        self.setup_precision_plugin()
```

Comment:
Does the order for these two matter? I see inconsistency across different strategies

Reply:
i prefer updating the optimizers first. setup_precision_plugin simply passes the model, optimizers, and lr_schedulers to the precision plugin to view